src/logic/context/ContextManager.ts

```diff
@@ -3,19 +3,35 @@ import {HotKeyAction} from "../../data/HotKeyAction";
 import {store} from "../../index";
 import {updateActiveContext} from "../../store/general/actionCreators";
 import * as _ from "lodash";
+import {EditorContext} from "./EditorContext";
+import {PopupContext} from "./PopupContext";
 
 export class ContextManager {
     private static activeCombo: string[] = [];
     private static actions: HotKeyAction[] = [];
+    private static contextHistory: Context[] = [];
 
     public static init(): void {
         window.addEventListener("keydown", ContextManager.onDown);
         window.addEventListener("keyup", ContextManager.onUp);
     }
 
-    public static switchCtx(context: Context, actions: HotKeyAction[]): void {
+    public static switchCtx(context: Context): void {
         store.dispatch(updateActiveContext(context));
-        ContextManager.actions = actions;
+        switch (context) {
```

Comment:
to zrobić jakoś generycznie. jakimś polimorficznym czymś xD
zrób klasę bazową, i tam metodę abstrakcyjną, czy coś. 
albo mapę obiektów. 
generalnie żeby nie trzeba było tego tak tutaj pisać. 
wiadomo że to pierdoła ale wiesz... :)

Reply:
Nie mam jakiegoś sensownego pomysłu na to :/